codegen/generate.go

```diff
@@ -0,0 +1,89 @@
+package codegen
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// pathToHomeDir yields the path to the terraform-vault-provider
+// home directory on the machine on which it's running.
+// ex. /home/your-name/go/src/github.com/terraform-providers/terraform-provider-vault
+var pathToHomeDir = func() string {
+	repoName := "terraform-provider-vault"
+	wd, _ := os.Getwd()
+	pathParts := strings.Split(wd, repoName)
+	return pathParts[0] + repoName
+}()
+
+/*
+codeFilePath creates a directory structure inside the "generated" folder that's
+intended to make it easy to find the file for each endpoint in Vault, even if
+we eventually cover all >500 of them and add tests.
+	terraform-provider-vault/generated$ tree
+	.
+	├── datasources
+	│   └── transform
+	│       ├── decode
+	│       │   └── role_name.go
+	│       └── encode
+	│           └── role_name.go
+	└── resources
+		└── transform
+			├── alphabet
+			│   └── name.go
+			├── alphabet.go
+			├── role
+			│   └── name.go
+			├── role.go
+			├── template
+			│   └── name.go
+			├── template.go
+			├── transformation
+			│   └── name.go
+			└── transformation.go
+*/
+func codeFilePath(tmplType templateType, path string) string {
+	return stripCurlyBraces(fmt.Sprintf("%s/generated/%s%s.go", pathToHomeDir, tmplType.String(), path))
```

Comment:
```suggestion
	filename := fmt.Sprintf("%s%s.go", tmplType.String(), path)
	path := filepath.Join(pathToHomeDir, "generated", filename)
	return stripCurlyBraces(path)
```

Reply:
I had to slightly tweak this to keep tests passing because path is a variable name already passed in. https://github.com/terraform-providers/terraform-provider-vault/pull/738/commits/52e1328c533e05f2e864bc3ee23d859e2065264d